### PR TITLE
[Fix](Nereids) fix leading with cte and same subqueryalias name

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
## Proposed changes

fix leading with cte and same subqueryalias name
Example:
with tbl1 as select t1.c1 from t1
select tbl2.c2 from (select / * + leading(t2 tbl1) * / tbl1.c1, t2.c2 from tbl1 join t2) as tbl2 join t3;
Reason:
in this case, before getting analyzed preprocess would change subquery tbl2 to cte plan, and this cte plan should be in upper level cte plan, but not in logical result sink plan

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/PullUpSubqueryAliasToCTE.java [hunk 1]
```diff
@@ -79,6 +79,15 @@ public Plan visitLogicalCTE(LogicalCTE<? extends Plan> logicalCTE, StatementCont
                 subQueryAlias = new LogicalSubQueryAlias<>(subQueryAlias.getAlias(), newSubQueryAlias);
             }
         }
-        return visitChildren(this, logicalCTE, context);
+        Plan cte = visitChildren(this, logicalCTE, context);
+        if (!aliasQueries.isEmpty()) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/PullUpSubqueryAliasToCTE.java:83]: add test case

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/PullUpSubqueryAliasToCTE.java (hunk 1 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/PullUpSubqueryAliasToCTE.java:83]: done

**Review by englefly — APPROVED** [review]

**Comment by englefly** [comment]
run performance

**Pull request merged**